iso15118/secc/secc_settings.py

```diff
@@ -70,22 +87,43 @@ def load_envs(self, env_path: Optional[str] = None) -> None:
         # the protocols are listed here determines the priority (i.e. first list entry
         # has higher priority than second list entry). A list entry must be a member
         # of the Protocol enum
-        self.supported_protocols = [
-            Protocol.DIN_SPEC_70121,
-            Protocol.ISO_15118_2,
-            Protocol.ISO_15118_20_AC,
-            Protocol.ISO_15118_20_DC,
-        ]
+        protocols = env.list("PROTOCOLS", default=self.default_protocols)
+        self.load_requested_protocols(protocols)
 
         # Supported authentication options (named payment options in ISO 15118-2).
         # Note: SECC will not offer 'pnc' if chosen transport protocol is not TLS
         # Must be a list containing either AuthEnum members EIM (for External
         # Identification Means), PNC (for Plug & Charge) or both
-        self.supported_auth_options = [AuthEnum.EIM, AuthEnum.PNC]
+        auth_modes = env.list("AUTH_MODES", default=self.default_auth_modes)
+        self.load_requested_auth_modes(auth_modes)
 
         # Whether the charging station allows the EV to go into Standby (one of the
         # enum values in PowerDeliveryReq's ChargeProgress field). In Standby, the
         # EV can still use value-added services while not consuming any power.
         self.standby_allowed = env.bool("STANDBY_ALLOWED", default=False)
 
         env.seal()  # raise all errors at once, if any
+
+    def load_requested_protocols(self, read_protocols: Optional[List[str]]):
+        protocols = format_list(read_protocols)
+        self.supported_protocols = []
+        for protocol in protocols:
+            if protocol in self.default_protocols:
+                self.supported_protocols.append(Protocol[protocol])
+            else:
+                logger.debug(
+                    f"Unknown protocol {protocol}. Supported protocols:"
+                    f" {self.default_protocols}"
+                )
```

Comment:
you can do the following:
```python
self.supported_protocols = list(set(protocols).intersection(self.default_protocols)
if self.supported_protocols is None:
    raise ...
```

I think we shall raise an exception in case there are no protocols requested that the system actually supports

Reply:
Sure, done.